website/blog/2021-11-14-Presto-Cluster-Memory-Tuning.md

```diff
@@ -0,0 +1,70 @@
+---
+author: George Wang
+authorURL: https://www.linkedin.com/in/george-wang-9a5a46190/
+title: Presto Cluster Memory Tuning
+---
+
+##Presto Memory Management
+
+Presto's memory management system is carefully designed to achieve high performing query engine in a distributed system. It defines the memory into two categories: user memory and system memory.
+* The user memory is related to user data. For example, reading user data from cloud storage on AWS S3 bucket will occupy the corresponding memory. The amount of data is tightly related;
```

Comment:
I think this description relating user memory to data read from cloud storage in S3 is a bit confusing, since the user memory is completely unrelated to how the data is stored and also is not used for reads.  what is true here is  memory is generally proportional to the amount of data being processed there are basically 2 cases when you use user memory.  1. building hash tables for joins and final aggregations and window functions, 2. Sorting for order bys and windows.  It's memory that Presto expects the user to be able to reason about when they think about the space complexity of a query.